internal/pkg/vaultworker/password.go

```diff
@@ -0,0 +1,112 @@
+/*******************************************************************************
+ * Copyright 2018 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * @author: Tingyu Zeng, Dell
+ * @version: 1.0.0
+ *******************************************************************************/
+
+package vaultworker
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/dghubble/sling"
+	"github.com/sethvargo/go-password/password"
+)
+
+type UserPasswd struct {
+    User string
+    Passwd string
+}
+
+func CreateCredential() (string, error) {
+	pass, err := password.Generate(8, 4, 4, false, false)
+	if err != nil {
+		return "", err
+	}
+	return pass, nil
+}
+
+func CredentialInStore(config *tomlConfig, secretBaseURL string, credPath string, c *http.Client) (bool, error) {
```

Comment:
instead of returning a `bool` and and `error`, you should just return an `error`. The only place where you return `false, nil` is at the end of the function when it's clear the operation was reported by vault to not be incorrect via the status code check and somehow an empty json response happened. In this instance returning an error explaining that would be better, i.e. `return errors.New("response from vault credential upload is invalid and has length 0")`

Reply:
will be addressed in #77.

Reply:
this is a different issue from the one in #77, this is about code clarity, not about the usage of a password generation library

Reply:
In this case it is not an error but indicates that the passwords haven't be initialized, so initialization need to be called. I am trying to avoid using errors for all the cases.

Reply:
okay, this is fine then